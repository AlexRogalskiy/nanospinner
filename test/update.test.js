let { createSpinner } = require('../index.js')

let stdout = { out: '' }
stdout.write = symbols => {
  stdout.out += symbols
}

it('uses custom frames', () => {
  stdout.out = ''
  let spinner = createSpinner('#update', { stream: stdout })

  spinner.spin()
  spinner.spin()
  spinner.update({ frames: ['.', 'o', '0', '@', '*'], text: 'Change update' })
  spinner.spin()
  spinner.spin()
  spinner.spin()

  expect(stdout.out).toMatchSnapshot(process.env.CI ? 'CI' : 'Local')
})
