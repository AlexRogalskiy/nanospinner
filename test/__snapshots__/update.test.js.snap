// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`uses custom frames: CI 1`] = `
"[33m-[39m #update
[33m-[39m #update
[33m.[39m Change update
[33mo[39m Change update
[33m0[39m Change update
"
`;

exports[`uses custom frames: Local 1`] = `"[?25l[1G[33mâ ‹[39m #update[?25l[1G[2K[1G[33mâ ™[39m #update[?25l[1G[2K[1G[33m0[39m Change update[?25l[1G[2K[1G[33m@[39m Change update[?25l[1G[2K[1G[33m*[39m Change update"`;
